plugins/tds40.js

```diff
@@ -553,38 +553,72 @@ tds = {
         }
     },
 
+    // Untangle TDS attributes & OSM tags.
+    // Some people have been editing OSM files and inserting TDS attributes
+    untangleAttributes: function (attrs, tags)
+    {
+        // If we use ogr2osm, the GDAL driver jams ant tag it doesn't know about into an "other_tags" tag.
+        // We need to unpack this before we can do anything.
+        if (attrs.other_tags)
+        {
+            var tList = attrs.other_tags.split('","');
+
+            delete attrs.other_tags;
+
+            for (var val in tList)
+            {
+                vList = tList[val].split('"=>"');
+
+                attrs[vList[0].replace('"','')] = vList[1].replace('"','');
+
+                // Debug
+                //print('val: ' + tList[val] + '  vList[0] = ' + vList[0] + '  vList[1] = ' + vList[1]);
+            }
+        }
+
+        for (var col in attrs)
+        {
+            // Find an FCODE
+            if (col in tds.fcodeLookup['F_CODE'])
```

Comment:
Unfortunately, when I was untangling the MGCP data I found this situation:
```
In Layername: lines
In Attrs: osm_id: :392107726:
In Attrs: other_tags: :"BH140"=>"river","CDA"=>"1000","GEOM"=>"GM_Curve","HYP"=>"1","SDP"=>"Digital_Globe","SDV"=>"2016-08-02","SRT"=>"30","TID"=>"1000","WCC"=>"7","WID"=>"2","WST"=>"1","security:classification"=>"UNCLASSIFIED","source:date"=>"2016-08-09":
In Attrs: waterway: :river:
Untangle Attrs: CDA: :1000:
Untangle Attrs: F_CODE: :BH140:
Untangle Attrs: HYP: :1:
Untangle Attrs: SDP: :Digital_Globe:
Untangle Attrs: SDV: :2016-08-02:
Untangle Attrs: SRT: :30:
Untangle Attrs: TID: :1000:
Untangle Attrs: WCC: :7:
Untangle Attrs: WID: :2:
Untangle Attrs: WST: :1:
``` 

The good thing is whoever was editing the data _did_ store the FCODE. The bad part is they stored it as "BH140=river".